File: BigFootVentures/BigFootVentures.Business.Objects/Management/Office.cs
using System;

namespace BigFootVentures.Business.Objects.Management
{
    public sealed class Office : BusinessBase
    {
        #region "Private Members"

        private string _registrationPaymentNotification;
        private string _registrationPaymentMethod;
        private string _oppositionPaymentNotification;
        private string _oppositionPaymentMethod;

        #endregion

        #region "Properties"

        public string OwnerName { get; set; }

        public string OfficeName { get; set; }
        public string ExternalID { get; set; }
        public string GeographyType { get; set; }
        public string State { get; set; }
        public string StateAbbreviation{ get; set; }
        public string OfficeProperty { get; set; }
        public string OfficeUrl { get; set; }
        public string OfficeValueCategory { get; set; }
        public bool IsOnline { get; set; }
        public string OnlineComments { get; set; }
        //public string CountryManager { get; set; }
        public string PublicationDateComments { get; set; }
        public string RegistrationDateComments { get; set; }
        public string OppositionPeriodMonths { get; set; }
        public string SearchUrl { get; set; }
        public string OfficeStatusesSource { get; set; }
        public string OfficeNameArchive { get; set; }
        public bool NationalNumberAssigned { get; set; }
        public string Points { get; set; }
        public bool PCT { get; set; }
        public bool Paris { get; set; }
        public bool WTO { get; set; }

        public string RegistrationPaymentNotification { get { return this._registrationPaymentNotification; } set { this._registrationPaymentNotification = value; if (!string.IsNullOrWhiteSpace(value)) { this.RegistrationPaymentNotifications = value.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries); } } }
        public string RegistrationPaymentMethod { get { return this._registrationPaymentMethod; } set { this._registrationPaymentMethod = value; if (!string.IsNullOrWhiteSpace(value)) { this.RegistrationPaymentMethods = value.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries); } } }
        public string OppositionPaymentNotification { get { return this._oppositionPaymentNotification; } set { this._oppositionPaymentNotification = value; if (!string.IsNullOrWhiteSpace(value)) { this.OppositionPaymentNotifications = value.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries); } } }
        public string OppositionPaymentMethod { get { return this._oppositionPaymentMethod; } set { this._oppositionPaymentMethod = value; if (!string.IsNullOrWhiteSpace(value)) { this.OppositionPaymentMethods = value.Split(new char[] { ';' }, StringSplitOptions.RemoveEmptyEntries); } } }

        #endregion

        #region "Calculated Properties"

        public string[] RegistrationPaymentNotifications { get; set; }
        public string[] RegistrationPaymentNotificationsAvailable { get; set; }
        public string[] RegistrationPaymentNotificationsSelected { get; set; }
        public string[] RegistrationPaymentMethods { get; set; }
        public string[] RegistrationPaymentMethodsAvailable { get; set; }
        public string[] RegistrationPaymentMethodsSelected { get; set; }
        public string[] OppositionPaymentNotifications { get; set; }
        public string[] OppositionPaymentNotificationsAvailable { get; set; }
        public string[] OppositionPaymentNotificationsSelected { get; set; }
        public string[] OppositionPaymentMethods { get; set; }
        public string[] OppositionPaymentMethodsAvailable { get; set; }
        public string[] OppositionPaymentMethodsSelected { get; set; }

        #endregion
    }
}

File: BigFootVentures/BigFootVentures.Application.Web/Scripts/Views/Shared/_Components/Modals/AdvanceSearch/_Office.js
function BuildOfficeQuery(query) {
    var officeName = $('[name="Input.Filter.Office.OfficeName"]').val();
    var geographyType = $('[name="Select.Filter.Office.GeographyType"]').val();
    var state = $('[name="Input.Filter.Office.State"]').val();
    var office = $('[name="Input.Filter.Office.Office"]').val();
    var officeURL = $('[name="Input.Filter.Office.OfficeURL"]').val();
    var officeValueCategory = $('[name="Select.Filter.Office.OfficeValueCategory"]').val();
    var online = $('[name="Select.Filter.Office.Online"]').val();
    var searchURL = $('[name="Input.Filter.Office.SearchURL"]').val();
    var officeNameArchive = $('[name="Input.Filter.Office.OfficeNameArchive"]').val();
    var nationalNumberAssigned = $('[name="Select.Filter.Office.Online"]').val();
    var PCT = $('[name="Select.Filter.Office.PCT"]').val();
    var paris = $('[name="Select.Filter.Office.Paris"]').val();
    var WTO = $('[name="Select.Filter.Office.WTO"]').val();

    if (officeName != null && officeName != undefined && officeName.length > 0) {
        query += '&officeName=' + officeName;
    }

    if (geographyType != '') {
        query += '&geographyType=' + geographyType;
    }

    if (state != null && state != undefined && state.length > 0) {
        query += '&state=' + state;
    }

    if (office != null && office != undefined && office.length > 0) {
        query += '&office=' + office;
    }

    if (officeURL != null && officeURL != undefined && officeURL.length > 0) {
        query += '&officeURL=' + officeURL;
    }

    if (officeName != null && officeName != undefined && officeName.length > 0) {
        query += '&officeName=' + officeName;
    }

    if (officeValueCategory != '') {
        query += '&officeValueCategory=' + officeValueCategory;
    }

    if (online != 'NotSet') {
        query += '&online=' + online;
    }

    if (searchURL != null && searchURL != undefined && searchURL.length > 0) {
        query += '&searchURL=' + searchURL;
    }

    if (officeNameArchive != null && officeNameArchive != undefined && officeNameArchive.length > 0) {
        query += '&officeNameArchive=' + officeNameArchive;
    }

    if (nationalNumberAssigned != 'NotSet') {
        query += '&nationalNumberAssigned=' + nationalNumberAssigned;
    }

    if (PCT != 'NotSet') {
        query += '&PCT=' + PCT;
    }

    if (paris != 'NotSet') {
        query += '&paris=' + paris;
    }

    if (WTO != 'NotSet') {
        query += '&WTO=' + WTO;
    }

    return query;
}